src/main/java/org/folio/inventory/instanceingress/handler/CreateInstanceIngressEventHandler.java

```diff
@@ -0,0 +1,240 @@
+          AdditionalFieldsUtil.addFieldToMarcRecord(marcBibRecord, TAG_035, SUBFIELD_A, event.getEventPayload().getSourceRecordIdentifier() + BIBFRAME);
+
+          LOGGER.info("Mapping a Record from InstanceIngressEvent with id '{}' into an Instance", event.getId());
+          var parsedRecord = new JsonObject((String) marcBibRecord.getParsedRecord().getContent());
+          RecordMapper<org.folio.Instance> recordMapper = RecordMapperBuilder.buildMapper(MARC_BIB_RECORD_FORMAT);
+          var instance = recordMapper.mapRecord(parsedRecord, mappingParameters, new JsonObject(mappingMetadata.getMappingRules()));
+          LOGGER.info("Mapped Instance from InstanceIngressEvent with id '{}': {}", event.getId(), instance);
+          return super.idStorageService.store(event.getId(), instance.getId(),
+              context.getTenantId())
+            .map(r -> instance)
+            .onFailure(e -> LOGGER.error("Error creating relationship of inventory recordId '{} and instanceId '{}'", event.getId(), instance.getId()));
+        } catch (Exception e) {
+          LOGGER.warn("Error during preparing and executing mapping:", e);
+          return Future.failedFuture(e);
+        }
+      });
+  }
+
+  private Future<Instance> validateInstance(org.folio.Instance instance, InstanceIngressEvent event) {
+    try {
+      LOGGER.info("Validating Instance from InstanceIngressEvent with id '{}':", event.getId());
+      var instanceAsJson = prepareInstance(instance, event.getEventPayload().getSourceType());
+      var errors = EventHandlingUtil.validateJsonByRequiredFields(instanceAsJson, INSTANCE_REQUIRED_FIELDS);
+      return failIfErrors(errors, event.getId())
+        .orElseGet(() -> {
+          var mappedInstance = Instance.fromJson(instanceAsJson);
+          var uuidErrors = ValidationUtil.validateUUIDs(mappedInstance);
+          return failIfErrors(uuidErrors, event.getId())
+            .orElseGet(() -> Future.succeededFuture(mappedInstance));
+        });
+    } catch (Exception e) {
+      return Future.failedFuture(e);
+    }
+  }
+
+  private static Optional<Future<Instance>> failIfErrors(List<String> errors, String eventId) {
+    if (!errors.isEmpty()) {
+      var msg = format("Mapped Instance is invalid: %s, from InstanceIngressEvent with id '%s' ", errors, eventId);
+      LOGGER.warn(msg);
+      return Optional.of(Future.failedFuture(msg));
+    }
+    return Optional.empty();
+  }
+
+  private JsonObject prepareInstance(org.folio.Instance instance, SourceType sourceType) {
+    var instanceAsJson = JsonObject.mapFrom(instance);
+    instanceAsJson.put(SOURCE_KEY, sourceType.value());
+    if (isNull(instanceAsJson.getString(INSTANCE_TYPE_ID_KEY))) {
+      //q is that ok?
+      LOGGER.warn("No instance type provided in mapped instance with id '{}', setting 'unspecified'", instance.getId());
+      instanceAsJson.put(INSTANCE_TYPE_ID_KEY, "30fffe0e-e985-4144-b2e2-1e8179bdb41f");
```

Comment:
What is this hardcoded ID? ( "30fffe0e-e985-4144-b2e2-1e8179bdb41f"). 
I guess this is the ID for type "unspecified". I guess this value will be different in different tenant databases and we cannot hardcode this in the code.

May be change the marc4ld library to always set and instancetype in the MARC record.

Reply:
yes, have tested the case when our library puts 008 anyway - and it's successful case, we don't need this hardcode anymore. Removed